utils/sv_transform_helper.py

```diff
@@ -0,0 +1,146 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+import bpy
+from bpy.props import (EnumProperty, BoolProperty)
+
+from sverchok.data_structure import updateNode
+from sverchok.settings import get_params
+
+from math import pi
+
+euler_order_items = [
+    ('XYZ', 'XYZ', "", 0),
+    ('XZY', 'XZY', "", 1),
+    ('YXZ', 'YXZ', "", 2),
+    ('YZX', 'YZX', "", 3),
+    ('ZXY', 'ZXY', "", 4),
+    ('ZYX', 'ZYX', "", 5)
+]
+
+
+class AngleUnits:
+    RADIANS = "RAD"
+    DEGREES = "DEG"
+    UNITIES = "UNI"
+
+    @classmethod
+    def get_blender_enum(cls):
+        return [
+            (AngleUnits.RADIANS, "Rad", "Radians", "", 0),
+            (AngleUnits.DEGREES, "Deg", "Degrees", "", 1),
+            (AngleUnits.UNITIES, "Uni", "Unities", "", 2)
+        ]
+
+
+angle_remap_options = {
+    # from               to
+    (AngleUnits.RADIANS, AngleUnits.DEGREES): 180/pi,
+    (AngleUnits.RADIANS, AngleUnits.UNITIES): 1/(2*pi),
+    (AngleUnits.DEGREES, AngleUnits.RADIANS): pi/180,
+    (AngleUnits.DEGREES, AngleUnits.UNITIES): 1/360,
+    (AngleUnits.UNITIES, AngleUnits.RADIANS): 2*pi,
+    (AngleUnits.UNITIES, AngleUnits.DEGREES): 360
+}
+
+
+class SvAngleHelper():
+
+    def get_preferences(self):
+        props = get_params({
+            'auto_update_angle_values': False,
+        })
+        return props.auto_update_angle_values
```

Comment:
I had originally added user default units via SV prefs here, but I couldn't figure out how to read those values from the prefs and updated them in the enum defaults for every node. (they only got updated once when registering the helper class. I removed the default units since but didn’t bother to change the method name as I though I’d add these later at some point in which case I would have had to make this a mixin method anyway (hence i kept the mixin name). But you’re right. I need to change this to perhaps get_auto_update_angle_values or something.